yammbs/torsion/_store.py

```diff
@@ -302,37 +310,79 @@ def optimize_mm(
                     ),
                 )
 
-    def get_log_sse(
+    def get_rmsd(
+        self,
+        force_field: str,
+        molecule_ids: list[int] | None = None,
+        skip_check: bool = False,
+    ) -> RMSDCollection:
+        """Get the RMSD summed over the torsion profile."""
+        from openff.toolkit import Molecule
+
+        if not molecule_ids:
+            molecule_ids = self.get_molecule_ids()
+
+        if not skip_check:
+            self.optimize_mm(force_field=force_field)
+
+        rmsds = RMSDCollection()
+
+        for molecule_id in molecule_ids:
+            if molecule_id not in molecule_ids:
+                continue
+
+            qm_points = self.get_qm_points_by_molecule_id(id=molecule_id)
+            mm_points = self.get_mm_points_by_molecule_id(id=molecule_id, force_field=force_field)
+
+            molecule = Molecule.from_mapped_smiles(
+                self.get_smiles_by_molecule_id(molecule_id),
+                allow_undefined_stereo=True,
+            )
+
+            rmsds.append(
+                RMSD(
+                    id=molecule_id,
+                    rmsd=sum(get_rmsd(molecule, qm_points[key], mm_points[key]) for key in qm_points),
+                ),
+            )
+
+        return rmsds
+
+    def get_rmse(
         self,
         force_field: str,
         molecule_ids: list[int] | None = None,
         skip_check: bool = False,
-    ) -> LogSSECollection:
+    ) -> RMSECollection:
+        """Get the RMSD summed over the torsion profile."""
+
         if not molecule_ids:
             molecule_ids = self.get_molecule_ids()
 
         if not skip_check:
             self.optimize_mm(force_field=force_field)
 
-        log_sses = LogSSECollection()
+        rmses = RMSECollection()
 
-        for molecule_id in self.get_molecule_ids():
+        for molecule_id in molecule_ids:
             if molecule_id not in molecule_ids:
                 continue
 
-            qm, mm = _normalize(
-                qm=self.get_qm_energies_by_molecule_id(molecule_id),
-                mm=self.get_mm_energies_by_molecule_id(molecule_id, force_field),
+            qm_energies, mm_energies = _normalize(
+                self.get_qm_energies_by_molecule_id(id=molecule_id),
+                self.get_mm_energies_by_molecule_id(id=molecule_id, force_field=force_field),
             )
 
-            log_sses.append(
-                LogSSE(
+            rmses.append(
+                RMSE(
                     id=molecule_id,
-                    value=math.log(sum([(mm[key] - qm[key]) ** 2 for key in qm])),
+                    rmse=numpy.linalg.norm(
+                        numpy.asarray([*qm_energies.values()]) - numpy.asarray([*mm_energies.values()]),
```

Comment:
I've seen this pattern before, but is there a benefit over something like `numpy.array(list(...))` or `numpy.fromiter` (which I just found out about while searching about this). Apparently you have to specify the `dtype` for `fromiter`, so maybe this is the tersest solution.

Reply:
Also I think this is not strictly the RMSE. The default norm is missing the "mean" part; it's just the root square error rather than root mean square. I think this is a fine metric on its own, but it might need a different name.

Reply:
I'll use EEN (energy error norm) as a stopgap

RMSD/RMSE/RMS is quite confusing, i.e. "RMSD" as commonly used doesn't explicitly say what it's measuring the deviation of

Reply:
re first comment: there is hysteresis here with a couple of different iterations and I think a mypy complaint at some point in time

I'll try `fromiter` on for size

Reply:
seems to work fine, and looks prettier IMHO